signalfx/resource_signalfx_team_links.go

```diff
@@ -0,0 +1,183 @@
+			d.SetId("")
+		}
+		return err
+	}
+
+	_, new := d.GetChange("detectors")
+	detectors := new.(*schema.Set)
+
+	dets := make([]interface{}, len(res.Detectors))
+	for i, detector := range res.Detectors {
+		dets[i] = detector
+	}
+
+	teamDets := schema.NewSet(schema.HashString, dets)
+	// XXX: For some reason the same sets are hashing to different values (maybe to interface{} vs string types).
+	// Rehash the set so intersection, etc. will work.
+	detectors = schema.NewSet(schema.HashString, detectors.List())
+
+	if err := d.Set("detectors", teamDets.Intersection(detectors)); err != nil {
+		return err
+	}
+
+	_, new = d.GetChange("dashboard_groups")
+	dashboardGroups := new.(*schema.Set)
+
+	dbgs := make([]interface{}, len(res.DashboardGroups))
+	for i, dbg := range res.DashboardGroups {
+		dbgs[i] = dbg
+	}
+
+	teamDbg := schema.NewSet(schema.HashString, dbgs)
+	// XXX: Rehash for same reason above.
+	dashboardGroups = schema.NewSet(schema.HashString, dashboardGroups.List())
+
+	if err := d.Set("dashboard_groups", teamDbg.Intersection(dashboardGroups)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func teamLinksCreate(d *schema.ResourceData, meta interface{}) error {
+	return teamLinksUpdate(d, meta)
+}
+
+func teamLinksUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*signalfxConfig)
+	team := d.Get("team").(string)
+
+	old, new := d.GetChange("detectors")
+	oldD := old.(*schema.Set)
+	newD := new.(*schema.Set)
+
+	removed := oldD.Difference(newD)
+	added := newD.Difference(oldD)
+
+	log.Printf("[DEBUG] SignalFx: Unlinking detectors from team %v: %v", team, removed.List())
+
+	for _, rd := range removed.List() {
+		det := rd.(string)
+		if err := config.Client.UnlinkDetectorFromTeam(context.TODO(), team, det); err != nil {
+			return err
+		}
+	}
+
+	log.Printf("[DEBUG] SignalFx: Linking detectors to team %v: %v", team, added.List())
+
+	for _, d := range added.List() {
+		det := d.(string)
+		if err := config.Client.LinkDetectorToTeam(context.TODO(), team, det); err != nil {
+			return err
+		}
+	}
+
+	if err := d.Set("detectors", new); err != nil {
+		return err
+	}
+
+	old, new = d.GetChange("dashboard_groups")
+	oldDbg := old.(*schema.Set)
+	newDbg := new.(*schema.Set)
+
+	removed = oldDbg.Difference(newDbg)
+	added = newDbg.Difference(oldDbg)
+
+	log.Printf("[DEBUG] SignalFx: Unlinking dashboard groups from team %v: %v", team, removed.List())
+
+	for _, d := range removed.List() {
+		det := d.(string)
+		if err := config.Client.UnlinkDashboardGroupFromTeam(context.TODO(), team, det); err != nil {
+			return err
+		}
+	}
+
+	log.Printf("[DEBUG] SignalFx: Linking dashboard groups to team %v: %v", team, added.List())
+
+	for _, d := range added.List() {
+		det := d.(string)
+		if err := config.Client.LinkDashboardGroupToTeam(context.TODO(), team, det); err != nil {
+			return err
+		}
+	}
+
+	if err := d.Set("dashboard_groups", new); err != nil {
+		return err
+	}
+
+	d.SetId("0")
```

Comment:
What does an id of `"0"` mean?

Reply:
Just a dummy value to signal to TF that the state should be saved.